srcopsmetrics/visualization.py

```diff
@@ -300,275 +287,278 @@ def visualize_results(project: str):
+
+        top_labels_list = sorted(top_labels_list, key=lambda tup: tup[2], reverse=True)
+        top_x_labels = top_labels_list[:top_x]
+
+        df = pd.DataFrame()
+        df["developer_label"] = [f"{label[0]} -> {label[1]}" for label in top_x_labels]
+        df["count"] = [label[2] for label in top_x_labels]
+
+        action = self._DEVELOPER_ACTION[developer_action]
+
+        fig = px.bar(
+            df,
+            x="developer_label",
+            y="count",
+            title=f"Top {top_x} overall {action} issue types in form of <{developer_action}> -> <issue_label>",
+            color="developer_label",
+        )
+        fig.show()
+
+    def _visualize_top_X_issues_types_wrt_project(
+        self, overall_types_data: Dict, developer_action: str, top_x: int = 5
+    ):
+        """Visualize overall top X issue types that had been opened || closed for given repository w.r.t to project."""
+        top_labels_list = {}
+        for author in overall_types_data.keys():
+            for label in overall_types_data[author].keys():
+                if label not in top_labels_list:
+                    top_labels_list[label] = 0
+                top_labels_list[label] += overall_types_data[author][label]
+
+        top_x_labels = zip([label for label in top_labels_list.keys()], [count for count in top_labels_list.values()])
+        top_x_labels = sorted(top_x_labels, reverse=True, key=lambda tup: tup[1])
+
+        df = pd.DataFrame()
+        df["label"] = [tup[0] for tup in top_x_labels][:top_x]
+        df["count"] = [tup[1] for tup in top_x_labels][:top_x]
+
+        action = self._DEVELOPER_ACTION[developer_action]
+
+        fig = px.bar(
+            df, x="label", y="count", title=f"Top {top_x} overall {action} issue types for project", color="label"
+        )
+        fig.show()
+
+    @staticmethod
+    def _visualize_ttci_wrt_pr_length(issues_data: IssueSchema, pr_data: PullRequestSchema) -> None:
+        """For each pull request size label visualize its TTCI.
+
+        Time To Close Issue is summed with respect to all of the Issues
+        the Pull requests with given size label have closed.
+
+        :param issues_data:IssueSchema:
+        :param pr_data:PullRequestSchema:
+        :rtype: None
+        """
+        pr_size_issues = pre_processing.pre_process_issues_closed_by_pr_size(issues_data=issues_data, pr_data=pr_data)
+        df = pd.DataFrame()
+
+        sizes = []
+        ttcis = []
+        for size in pr_size_issues.keys():
+            for ttci in pr_size_issues[size]:
+                sizes.append(size)
+                ttcis.append(ttci)
+
+        df["size"] = sizes
+        df["TTCI"] = ttcis
+
+        fig = px.scatter(df, x="size", y="TTCI", color="size")
+        fig.show()
+
+    @staticmethod
+    def _visualize_ttci_wrt_labels(
+        self, issues_data: IssueSchema, statistical_quantity: str = StatisticalQuantityEnum.MEDIAN.value
+    ) -> None:
```

Comment:
should not be static probably (use of self)

Reply:
Addressed in https://github.com/AICoE/SrcOpsMetrics/pull/54/commits/35edb3976c9736ccfe4e9f7421f811af0f827571! Thanks :)